src/RobotMarkerGenerator.cpp

```diff
@@ -50,32 +65,46 @@ visualization_msgs::Marker RobotMarkerGenerator::createMarker(string frameId)
     return marker;
 }
 
-visualization_msgs::Marker RobotMarkerGenerator::createMarkerLabel(string frameId, string marker_label)
+visualization_msgs::Marker RobotMarkerGenerator::createMarkerLabel(
+    string frameId, 
+    string marker_label
+)
 {
     visualization_msgs::Marker marker;
 
     marker = createMarker(frameId);
     marker.type = visualization_msgs::Marker::TEXT_VIEW_FACING;
 
+    marker.id = marker.id + 1000;
```

Comment:
what if we have `1001` joint ?

Reply:
I'm not sure if I get the question right. Anyway, what I'm doing here is to add a fixed number to the previously generated id so the individual markers within the marker-array have distinct ids. This is necessary since `createMarker()` creates a marker with a fixed id calculated from the given frame's name. As it is called multiple times for the same frame, the markers in the array would overwrite each other otherwise.

Reply:
no. i meant something else, but its okey